tensorflow/contrib/eager/python/examples/generative_examples/dcgan.ipynb

```diff
@@ -84,114 +192,135 @@
         "train_dataset = tf.data.Dataset.from_tensor_slices(train_images).shuffle(BUFFER_SIZE).batch(BATCH_SIZE)"
-      ]
+      ],
+      "execution_count": 0,
+      "outputs": []
     },
     {
+      "metadata": {
+        "id": "THY-sZMiQ4UV",
+        "colab_type": "text"
+      },
       "cell_type": "markdown",
+      "source": [
+        "## Create the models\n",
+        "\n",
+        "We will use tf.keras model subclassing to create the generator and discriminator. We will create layers in the __init__ method and set them as attributes of the class instance. And then define the forward pass in the **call **method."
+      ]
+    },
+    {
       "metadata": {
-        "colab_type": "text",
-        "id": "THY-sZMiQ4UV"
+        "id": "-tEyxE-GMC48",
+        "colab_type": "text"
       },
+      "cell_type": "markdown",
       "source": [
-        "## Write the generator and discriminator models\n",
+        "### The Generator Model\n",
         "\n",
-        "* **Generator** \n",
-        "  * It is responsible for **creating convincing images that are good enough to fool the discriminator**.\n",
-        "  * It consists of Conv2DTranspose (Upsampling) layers. We start with a fully connected layer and upsample the image 2 times so as to reach the desired image size (mnist image size) which is (28, 28, 1). \n",
-        "  * We use **leaky relu** activation except for the **last layer** which uses **tanh** activation.\n",
-        "  \n",
-        "* **Discriminator**\n",
-        "  * **The discriminator is responsible for classifying the fake images from the real images.**\n",
-        "  * In other words, the discriminator is given generated images (from the generator) and the real MNIST images. The job of the discriminator is to classify these images into fake (generated) and real (MNIST images).\n",
-        "  * **Basically the generator should be good enough to fool the discriminator that the generated images are real**."
+        "The **generator **is responsible for **creating convincing images that are good enough to fool the discriminator**. \n",
+        "\n",
+        "Here is the network architecture for the generator:\n",
+        " * It consists of Conv2DTranspose (Upsampling) layers. We start with a fully connected layer and **upsample** the image 2 times in order to reach the desired image size as mnist image size of (28, 28, 1). We increase the width and height, and reduce the depth as we move through the layers in the network.\n",
+        " * We use **leaky relu** activation except for the **last layer** which uses **tanh** activation."
       ]
     },
     {
-      "cell_type": "code",
-      "execution_count": 0,
       "metadata": {
-        "colab": {},
+        "id": "VGLbvBEmjK0a",
         "colab_type": "code",
-        "id": "VGLbvBEmjK0a"
+        "colab": {}
       },
-      "outputs": [],
+      "cell_type": "code",
       "source": [
         "class Generator(tf.keras.Model):\n",
```

Comment:
Both this and the discriminator class seems like they will be well served by `tf.keras.Sequential`, making the code even more compact. Should we write them instead as:

```python
def make_generator_model():
  return tf.keras.Sequential([
    tf.keras.layers.Dense(7*7*64, use_bias=False),
    tf.keras.layers.BatchNormalization(),
    tf.keras.layers.Relu(),
    ...])
```